api/config/middleware/v1/middleware.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+
+package clutch.config.middleware.v1;
+
+option go_package = "github.com/lyft/clutch/backend/api/config/middleware/v1;middlewarev1";
+
+import "google/protobuf/any.proto";
+import "validate/validate.proto";
+
+message AccessLog {
+  // filter on an interger comparison or range of values
+  message StatusCodeFilter {
+    enum Op {
+      UNSPECIFIED = 0;
+      // =
+      EQ = 1;
+      // >=
+      GE = 2;
+      // <=
+      LE = 3;
+      // !=
+      NE = 4;
+    }
+
+    // Comparison operator.
+    Op operator = 1 [ (validate.rules).enum = {defined_only : true} ];
+    // gRPC status code value
+    uint32 value = 2;
+  }
+  // if no filter is specified all status codes are logged
+  StatusCodeFilter status_code_filter = 1;
```

Comment:
Oh ok the reason I took the operator approach was to avoid the `repeated` field assuming a user would only want to filter based on a single status code value. I see that something like `log status codes == 3 OR == 9` would not be possible in this case. Will switch up the design to what you suggested.

Reply:
Would it be easier to just have a `notequal` field in filterType so the user doesn't have to represent !=X as range values?

Reply:
we could, though it may be confusing to have a filter with a built-in negation because the filter's are OR'd together

so filters: `[ne: 1, range:[0,15]]`, would still log 1 since it matches the second filter.

honestly we can probably get away with equals for this entire PR... there are only 16 codes for gRPC (compared to HTTP which has 30 or more and meaningful ranges). even if i have to do `eq: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]` to log everything but 0 it's not that bad